### PR TITLE
Add IBM Granite Embedding Models

### PR DESCRIPTION
**Pull request by aashka-trivedi**
## Checklist

- [x] Run tests locally to make sure nothing is broken using `make test`. 
- [x] Run the formatter to format the code using `make lint`. 

### Adding a model checklist

 - [x] I have filled out the ModelMeta object to the extent possible
 - [x] I have ensured that my model can be loaded using
   - [x] `mteb.get_model(model_name, revision)` and
   - [x] `mteb.get_model_meta(model_name, revision)`
 - [x] I have tested the implementation works on a representative set of tasks.

### CODE REVIEW

File: mteb/models/ibm_granite_models.py [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+    "jpn_Latn",
+    "kor_Latn",
+    "nld_Latn",
+    "por_Latn",
+    "zho_Hant",
+    "zho_Hans",
+]
+
+
+granite_107m_multilingual = ModelMeta(
```
Comment (isaac-chung) [inline mteb/models/ibm_granite_models.py:24]: Could you please fill out adapted_from and superseded_by in the [ModelMeta](https://github.com/embeddings-benchmark/mteb/blob/48cb97d1e75cdf8260283d6628ac515141ed1e92/mteb/model_meta.py#L81) as well? (same for all below).
```
adapted_from: Name of the model from which this model is adapted from. For quantizations, fine-tunes, long doc extensions, etc.
```

**Review by isaac-chung — COMMENTED** [review]
Thanks for adding this! Just a small comment. Looks good otherwise.

File: mteb/models/ibm_granite_models.py (hunk 1 above)
Comment (aashka-trivedi) [reply mteb/models/ibm_granite_models.py:24]: Thanks for the review! The base model is not a public model (they are RoBERTa-like models we trained internally at IBM Research), which is why I left that field empty. Do you have a recommendation of what I should add here, considering it cannot be accessed? Ty!

Comment (isaac-chung) [reply mteb/models/ibm_granite_models.py:24]: `None` is fine here for both fields :)

Comment (aashka-trivedi) [reply mteb/models/ibm_granite_models.py:24]: Makes sense, Done!

**Review by isaac-chung — APPROVED** [review]
Thanks again! Will merge.

**Pull request merged**